forte/processors/data_augment/algorithms/UDA.py

```diff
@@ -0,0 +1,192 @@
+            prediction classes.
+        - softmax_temperature:
+            The softmax temperature for sharpening the distribution.
+        - confidence_threshold:
+            The threshold for confidence-masking. It is a threshold
+            of the probability in [0, 1], rather than of the logit.
+            If set to -1, the threshold will be ignored.
+        - reduction:
+            Default: 'mean'.
+            This is the same as the `reduction` argument in :func:
+            texar.losses.info_loss.kl_divg_loss_with_logits.
+            Specifies the reduction to apply to the output:
+
+            - :attr:`'none'`: no reduction will be applied.
+            - :attr:`'batchmean'`: the sum of the output will be
+                divided by the batchsize.
+            - :attr:`'sum'`: the output will be summed.
+            - :attr:`'mean'`: the output will be divided by
+                the number of elements in the output.
+    """
+    def __init__(
+            self,
+            sup_iterator: DataIterator,
+            unsup_iterator: DataIterator,
+            unsup_forward_fn: Callable,
+            softmax_temperature: float = 1.0,
+            confidence_threshold: float = -1,
+            reduction: str = "mean",
+    ):
+        self.sup_iterator: DataIterator = sup_iterator
+        self.unsup_iterator: DataIterator = unsup_iterator
+        self.unsup_forward_fn = unsup_forward_fn
+        self.softmax_temperature = softmax_temperature
+        self.confidence_threshold = confidence_threshold
+        self.reduction = reduction
+
+        # The flag for returning the unsupervised data.
+        self.use_unsup = True
+        self.sup_iter: Iterator[Batch]
+        self.unsup_iter: Iterator[Batch]
+
+    def __len__(self):
+        return self.sup_iterator.__len__()
+
+    def switch_to_dataset(
+            self,
+            dataset_name: Optional[str] = None,
+            use_unsup: bool = True
+    ):
+        # Set the flag of using unsupervised data.
+        self.use_unsup = use_unsup
+        self.sup_iterator.switch_to_dataset(dataset_name)
+
+    def switch_to_dataset_unsup(
+            self,
+            dataset_name: Optional[str] = None
+    ):
+        self.unsup_iterator.switch_to_dataset(dataset_name)
+
+    @property
+    def num_datasets(self) -> int:
+        return self.sup_iterator.num_datasets \
+               + self.unsup_iterator.num_datasets
+
+    @property
+    def dataset_names(self) -> List[str]:
+        return self.sup_iterator.dataset_names \
+               + self.unsup_iterator.dataset_names
+
+    def calculate_uda_loss(
+            self,
+            unsup_batch: Batch,
+            unsup_forward_fn: Callable
+    ) -> Tensor:
+        r"""
+        This function calculate the KL divergence
+        between the output probabilities of original
+        input and augmented input. It calls the user-provided
+        :func: unsup_forward_fn to perform the prediction.
```

Comment:
```suggestion
        :func:`unsup_forward_fn` to perform the prediction.
```

I think we need to use quotes around the function name.